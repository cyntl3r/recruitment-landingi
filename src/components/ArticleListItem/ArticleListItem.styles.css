.item {
  padding: 10px 0;
  border-bottom: 1px solid #dddddd;
}

.item:last-child {
  border-bottom: none;
}

.link {
  color: #000000;
  text-decoration: none;
  font-size: 1.2rem;
  display: flex;
  align-items: center;
}

.number {
  font-size: 0.8rem;
  width: 45px;
  color: #dddddd;
  padding-top: 1px;
}
