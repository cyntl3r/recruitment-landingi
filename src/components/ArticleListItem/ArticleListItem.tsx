import { FunctionComponent } from 'react';
import { Link } from 'react-router-dom';
import { isArticleInFavourites } from '../../utils/articleFavourites';
import styles from './ArticleListItem.styles.css';

export interface ArticleListItemProps {
  id: number;
  title: string;
}

export const ArticleListItem: FunctionComponent<ArticleListItemProps> = ({
  id,
  title,
}) => (
  <li data-testid="article-list-item" className={styles.item}>
    <Link to={`/article/${id}`} className={styles.link}>
      <span className={styles.number}>#{id}</span> {title}{' '}
      {isArticleInFavourites(id) && '❤️'}
    </Link>
  </li>
);
