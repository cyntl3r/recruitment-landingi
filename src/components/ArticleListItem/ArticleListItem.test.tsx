import { render } from '../../testing/test-utils';
import { ArticleListItem } from './ArticleListItem';

test('snapshot', () => {
  const { container } = render(<ArticleListItem id={2} title="test" />);
  expect(container).toMatchSnapshot();
});

test('wrapper render', () => {
  const { getByTestId } = render(<ArticleListItem id={1} title="test" />);
  expect(getByTestId('article-list-item')).toBeDefined();
});
