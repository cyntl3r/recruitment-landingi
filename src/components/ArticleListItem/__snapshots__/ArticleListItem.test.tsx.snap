// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div>
  <li
    data-testid="article-list-item"
  >
    <a
      href="/article/2"
    >
      <span>
        #
        2
      </span>
       
      test
       
    </a>
  </li>
</div>
`;
